internal/middleware/account_test.go

```diff
@@ -0,0 +1,35 @@
+package middleware_test
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
+	"github.com/RHEnVision/provisioning-backend/internal/dao/stubs"
+	"github.com/RHEnVision/provisioning-backend/internal/middleware"
+	"github.com/RHEnVision/provisioning-backend/internal/testing/identity"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAccountMiddleware(t *testing.T) {
+	ctx := identity.WithIdentity(t, context.Background())
+	ctx = stubs.WithAccountDaoOne(ctx)
+
+	req, err := http.NewRequestWithContext(ctx, "GET", "/api/provisioning/v1/pubkeys", nil)
+	if err != nil {
+		panic(err)
+	}
+
+	rr := httptest.NewRecorder()
+
+	isAccInNext := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var acc = ctxval.GetAccount(r.Context())
+		assert.NotNil(t, acc, "account was not set")
+	})
```

Comment:
Can you pls explain why is it necessary to check `AccountMiddleware` function? 
Is it because we are expecting the account to be set here: https://github.com/ezr-ondrej/provisioning-backend/blob/5da17892cd4e228de458787bffecad7272f9684f/internal/middleware/account.go#L31 ?

Reply:
Yeah, basically :)
This test is testing whether the AccountMiddleware "had set the account into the context"

Given context is only passed to the next middleware in the stack, we create a function that is actually checking the value in the context and we add this function as a next function in the stack and thus there are two functions in the middleware stack: 1) AccountMiddleware 2) the check function

So AccountMiddleware should set account to context and send the context to the following middleware, which in here is the test function, so this fnc can verify what is actually in the context :)